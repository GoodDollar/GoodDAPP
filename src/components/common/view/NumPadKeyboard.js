// @flow
import React from 'react'
import { View } from 'react-native'
import { amountRegexp, isNativeToken, moneyRegexp } from '../../../lib/wallet/utils'
import { withStyles } from '../../../lib/styles'
import KeyboardKey from './KeyboardKey'
import KeyboardRow from './KeyboardRow'

type CaretPosition = {
  start: number,
  end: number,
}

type KeyboardProps = {
  onPress: string => void,
  amount: string,
  caretPosition?: CaretPosition,
  updateCaretPosition?: CaretPosition => void,
}

const NumPadKeyboard = ({
  onPress,
  amount,
  isMaxLength,
  caretPosition,
  updateCaretPosition,
  styles,
  unit = 'G$',
}: KeyboardProps) => {
  const regex = isNativeToken(unit) ? amountRegexp : moneyRegexp

  const onPressKey = (value: string) => {
    // prevent adding numbers to the amount field if maxLength is reached
    if (isMaxLength) {
      return
    }

    const stringAmount = `${amount}`
    const updatedValue = caretPosition
      ? [stringAmount.slice(0, caretPosition.start), value, stringAmount.slice(caretPosition.end)].join('')
      : `${stringAmount}${value}`

    if (regex.test(updatedValue)) {
      onPress(updatedValue)
      updateCaretPosition({
        start: caretPosition.start + 1,
        end: caretPosition.start + 1,
      })
    }
  }

  const onBackspaceKey = () => {
    if (!caretPosition || caretPosition.end > 0) {
      const stringAmount = `${amount}`
      let updatedValue = stringAmount.slice(0, -1)
      if (caretPosition) {
        updatedValue = [stringAmount.slice(0, caretPosition.start - 1), stringAmount.slice(caretPosition.end)].join('')
        updateCaretPosition({
          start: caretPosition.start - 1,
          end: caretPosition.start - 1,
        })
      }
      onPress(updatedValue)
    }
  }

  return (
    <View style={styles.keyboard}>
      <KeyboardRow keys={['1', '2', '3']} onPress={onPressKey} />
      <KeyboardRow keys={['4', '5', '6']} onPress={onPressKey} />
      <KeyboardRow keys={['7', '8', '9']} onPress={onPressKey} />
      <View style={styles.row}>
        <KeyboardKey keyValue="." onPress={onPressKey} />
        <KeyboardKey keyValue="0" onPress={onPressKey} />
        <KeyboardKey keyValue="backspace" onPress={onBackspaceKey} />
      </View>
    </View>
  )
}

NumPadKeyboard.defaultProps = {
  caretPosition: null,
  updateCaretPosition: () => {},
}

const getStylesFromProps = ({ theme }) => {
  return {
    keyboard: {
      display: 'flex',
      marginBottom: theme.sizes.defaultDouble,
      marginTop: theme.sizes.defaultDouble,
      maxHeight: 250,
      flexGrow: 1,
      overflow: 'hidden',
      justifyContent: 'space-between',
    },
    row: {
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-between',
    },
  }
}

export default withStyles(getStylesFromProps)(NumPadKeyboard)
