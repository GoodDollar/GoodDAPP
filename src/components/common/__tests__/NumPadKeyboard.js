import React from 'react'
import renderer from 'react-test-renderer'
import NumPadKeyboard from '../NumPadKeyboard'

// Note: test renderer must be required after react-native.

describe('NumPadKeyboard', () => {
  let amount = 14
  const handleAmountChange = value => {
    amount = value
  }

  it('renders without errors', () => {
    const tree = renderer.create(<NumPadKeyboard onPress={handleAmountChange} amount={amount} />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<NumPadKeyboard onPress={handleAmountChange} amount={amount} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
