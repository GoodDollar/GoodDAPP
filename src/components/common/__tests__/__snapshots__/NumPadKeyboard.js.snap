// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumPadKeyboard matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "display": "flex",
      "paddingBottom": "10px",
      "paddingLeft": "10px",
      "paddingRight": "10px",
      "paddingTop": "10px",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "justify",
        "WebkitFlexDirection": "row",
        "WebkitJustifyContent": "space-between",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "msFlexAlign": "center",
        "msFlexDirection": "row",
        "msFlexPack": "justify",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        1
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        2
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        3
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "justify",
        "WebkitFlexDirection": "row",
        "WebkitJustifyContent": "space-between",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "msFlexAlign": "center",
        "msFlexDirection": "row",
        "msFlexPack": "justify",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        4
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        5
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        6
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "justify",
        "WebkitFlexDirection": "row",
        "WebkitJustifyContent": "space-between",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "msFlexAlign": "center",
        "msFlexDirection": "row",
        "msFlexPack": "justify",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        7
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        8
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        9
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitAlignItems": "center",
        "WebkitBoxAlign": "center",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "horizontal",
        "WebkitBoxPack": "justify",
        "WebkitFlexDirection": "row",
        "WebkitJustifyContent": "space-between",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "msFlexAlign": "center",
        "msFlexDirection": "row",
        "msFlexPack": "justify",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        .
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-text-76zvg2"
        dir="auto"
        style={
          Object {
            "color": "rgba(66,69,74,1.00)",
            "fontFamily": "RobotoSlab-Bold",
            "fontSize": "20px",
            "fontWeight": 700,
          }
        }
      >
        0
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      data-focusable={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozUserSelect": "none",
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 1,
          "WebkitFlexBasis": "0%",
          "WebkitFlexGrow": 1,
          "WebkitFlexShrink": 1,
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "cursor": "pointer",
          "display": "flex",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "msFlexAlign": "center",
          "msFlexNegative": 1,
          "msFlexPositive": 1,
          "msFlexPreferredSize": "0%",
          "msTouchAction": "manipulation",
          "msUserSelect": "none",
          "paddingBottom": "15px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "15px",
          "touchAction": "manipulation",
          "userSelect": "none",
        }
      }
      tabIndex="0"
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "backgroundImage": "url(backKeyboardButton.png)",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "contain",
            "cursor": "pointer",
            "height": "25px",
            "width": "25px",
          }
        }
      />
    </div>
  </div>
</div>
`;
